src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -36,15 +43,38 @@ struct ddsrt_ifaddrs {
 
 typedef struct ddsrt_ifaddrs ddsrt_ifaddrs_t;
 
+/**
+ * @brief Get the interface addresses (@ref ddsrt_ifaddrs) for specific address families
+ * 
+ * Note that array 'afs' must be terminated with an element 'DDSRT_AF_TERM'.
+ * 
+ * @param[out] ifap interface address pointer
```

Comment:
This is a bit too vague.
* on success, `*ifap` will be point to a newly allocated list of interfaces — that it is allocated matters because it means you can don't have to free it before calling it again.
* on failure, `*ifap` is left untouched

it is also not simply "interface addresses", you get interface information, including addresses.